process_cohort.smk

```diff
@@ -42,6 +58,36 @@ else:
     singleton = False
 print(f"Samples in cohort: {samples}.")
 
+# find all trios in cohort
+trio_dict = get_trios()
+if trio_dict:
+    print(f"Parents listed for samples: {list(trio_dict.keys())}")
+    for s in trio_dict.keys():
+        print(f"Parent IDs for {s}: {trio_dict[s]['parent1']}, {trio_dict[s]['parent2']}")
+else:
+    print(f"No trios found in cohort.")
+
+# scan smrtcells/ready directory for uBAMs or FASTQs that are ready to process
+# uBAMs have priority over FASTQs in downstream processes if both are available
+ubam_pattern = re.compile(r'smrtcells/ready/(?P<sample>[A-Za-z0-9_-]+)/(?P<movie>m\d{5}[Ue]?_\d{6}_\d{6}).(ccs|hifi_reads).bam')
+ubam_dict = defaultdict(dict)
+fastq_pattern = re.compile(r'smrtcells/ready/(?P<sample>[A-Za-z0-9_-]+)/(?P<movie>m\d{5}[Ue]?_\d{6}_\d{6}).fastq.gz')
+fastq_dict = defaultdict(dict)
+ubam_fastq_dict = defaultdict(dict)
+for infile in Path('smrtcells/ready').glob('**/*.bam'):
+    ubam_match = ubam_pattern.search(str(infile))
+    if ubam_match:
+        # create a dict-of-dict to link samples to movie context to uBAM filenames
+        ubam_dict[ubam_match.group('sample')][ubam_match.group('movie')] = str(infile)
+        ubam_fastq_dict[ubam_match.group('sample')][ubam_match.group('movie')] = str(infile)
```

Comment:
What happens if both a uBAM and a FASTQ are available for the same movie?

Reply:
Looks like the answer is, maybe nothing?  If you only need the keys from `ubam_fastq_dict[sample]`, might be able to get them from `movie_dict` instead.

Reply:
Good point. I was trying to avoid dependency on outputs from process_sample

Reply:
Could probably grab the keys from `ubam_dict[sample]` and `fastq_dict[sample]` and merge them.